sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateProjectDao.java

```diff
@@ -0,0 +1,254 @@
+  }
+
+  @CoverageIgnore
+  public JdbcTemplate getJdbcTemplate() {
+    return template;
+  }
+  @Override
+  public Map<String, Integer> getProjectColumnSizes() throws IOException {
+    return DbUtils.getColumnSizes(template, TABLE_NAME);
+  }
+  @Override
+  @CoverageIgnore
+  public ProjectOverview getProjectOverviewById(long overviewId) throws IOException {
+    return (ProjectOverview) currentSession().get(ProjectOverview.class, overviewId);
+  }
+
+  private Group getProjectWatcherGroup() throws IOException {
+    return getSecurityStore().getGroupByName("ProjectWatchers");
+  }
+
+  public SecurityStore getSecurityStore() {
+    return securityStore;
+  }
+
+  public void setWatchManager(WatchManager watchManager) {
+    this.watchManager = watchManager;
+  }
+
+  @Override
+  public void addWatcher(Project project, User watcher) {
+    watchManager.watch(project, watcher);
+    currentSession().update(project);
+  }
+
+  @Autowired
+  private WatchManager watchManager;
+
+  @Override
+  public void removeWatcher(Project project, User watcher) {
+    watchManager.unwatch(project, watcher);
+    currentSession().update(project);
+  }
+
+  @Override
+  public Project lazyGet(long projectId) throws IOException {
+    return get(projectId);
+  }
+
+  public ProjectOverview lazyGetProjectOverviewById(long overviewId) throws IOException {
+    return getProjectOverviewById(overviewId);
+  }
+  @Override
+  public List<Project> listAll() throws IOException {
+    Criteria criteria = currentSession().createCriteria(ProjectImpl.class);
+    @SuppressWarnings("unchecked")
+    List<Project> results = criteria.list();
+    return withWatcherGroup(results);
+  }
+
+  @Override
+  public List<Project> listAllWithLimit(long limit) throws IOException {
+    Criteria criteria = currentSession().createCriteria(ProjectImpl.class);
+    criteria.setMaxResults((int) limit);
+    @SuppressWarnings("unchecked")
+    List<Project> results = criteria.list();
+    return withWatcherGroup(results);
+  }
+
+  @Deprecated
+  @Override
+  public List<Project> listBySearch(String query) throws IOException {
+    Criteria criteria = currentSession().createCriteria(ProjectImpl.class);
+    criteria.add(DbUtils.searchRestrictions(query, SEARCH_PROPERTIES));
+    @SuppressWarnings("unchecked")
+    List<Project> results = criteria.list();
+    return withWatcherGroup(results);
+  }
+
+  @Override
+  public List<ProjectOverview> listOverviewsByProjectId(long projectId) throws IOException {
+    Criteria criteria = currentSession().createCriteria(ProjectOverview.class);
+    criteria.createAlias("project", "project");
+    criteria.add(Restrictions.eq("project.id", projectId));
+    @SuppressWarnings("unchecked")
+    List<ProjectOverview> results = criteria.list();
+    return results;
+  }
+
+  public boolean removeOverview(ProjectOverview overview) throws IOException {
+    if (overview.isDeletable()) {
+      currentSession().delete(overview);
+      return true;
```

Comment:
If you've added a CascadeType to the Project's ProjectOverviews, the overview will first need to be dissociated from the project before delete.